lib/charms/acme_client_operator/v0/acme_client.py

```diff
@@ -0,0 +1,207 @@
+"""# acme_client Library.
+This library is designed to enable developers to easily create new charms for implementations of the ACME protocol.
+This library contains all the logic necessary to get certificates from an ACME server..
+
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+charmcraft fetch-lib charms.acme_client_operator.v0.acme_client
+```
+You will also need to add the following library to the charm's `requirements.txt` file:
+- jsonschema
+- cryptography
+
+Then, to use the library in an example charm, you can do the following:
+```python
+from charms.acme_client_operator.v0.acme_client import AcmeClient
+from ops.main import main
+class ExampleAcmeCharm(AcmeClient):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._server = "https://acme-staging-v02.api.letsencrypt.org/directory"
+    @property
+    def cmd(self):
+        return [
+            "lego",
+            "--email",
+            self._email,
+            "--accept-tos",
+            "--csr",
+            "/tmp/csr.pem",
+            "--server",
+            self._server,
+            "--dns",
+            "namecheap",
+            "run",
+        ]
```

Comment:
But in case we start using `certbot`  this command would have to change in the superclass? is that what we want? if yes, I'm happy to move it :)

Reply:
If we switch to `certbot`, the format of the certificate files might also change, and how you pass CSRs. I think all that logic should be hidden as much as possible from the subclasses, otherwise we will have to change all the subclasses for such changes.

Reply:
✅